Blog.API/Blog.Infrastructure/Abstract/UnitOfWork.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Blog.Infrastructure.Abstract.Interfaces;
+using Blog.Infrastructure.Data;
+using System.Threading.Tasks;
+
+namespace Blog.Infrastructure.Abstract;
+
+public sealed class UnitOfWork : IUnitOfWork
+{
+    private readonly ApiDataContext _context;
+
+    public UnitOfWork(ApiDataContext context)
+    {
+        _context = context;
+    }
+
+    public IGenericRepository<T> GenericRepository<T>() where T : class
+    {
+        return new GenericRepository<T>(_context);
+    }
```

Comment:
I don't think that creating class every time is memory efficient. This need to be reworked. This might work:
```suggestion
public IGenericRepository<T> GenericRepository => _genericRepository;
```